### ISSUE TITLE
Bug: Custom list route fails to render list if it contains emails with attachments

### ISSUE THREAD

**Issue by Roensby** [InboxSDK/InboxSDK#1130]
A custom route list handler correctly renders the list of emails. Then I click on a email message to view it, return to another view like SENT and when I come back to the custom list route, it fails to render, instead showing the "No messages matched your search" message. This only happens when the list contains an email with an attachment. If that message is deleted and not included in the list, the list renders every time without issue.

**Steps to reproduce**

1. Create a custom list route handler that returns a list of emails where at least one of them has an attachment

```
sdk.Router.handleCustomListRoute('my-custom-list-route', (offset, max) => {
  return new Promise((resolve, reject) => {
	const threadIds = ...
	resolve({
	  threads: threadIds,
	  total: threadIds.length,
	})
  })
})
```

2. Visit the list to confirm it works

3. View any message (this seems to force a re-render of the list)

4. Visit any other list (like SENT)

5. Revisit the list and observe that now it does not render

**Expected behavior**
The custom list renders correctly every time it is visited

**Version**
NPM 2.1.30

**Some observations**

- When the custom list route handler fails to render the list, it can be observed that `driver.getPageCommunicator().setCustomListResults()` still returns results. I have made a diff between the results the first time the list is visited (when it correctly renders) and the second time the list is visited (when it fails to render). The only difference between them are *timestamps* and *attachment IDs*. Apparently this is enough for the custom list to fail to render a second time.

- replaceThreadsInSearchResponse_20220909 processes the custom route list before it is rendered. If this method is circumvented, the list renders correctly (albeit unsorted) every time. This may indicate that `driver.getPageCommunicator().setCustomListNewQuery()` somehow returns results that do not render correctly the second time the custom list is visited.

**Comment by Roensby** [comment]
This seems harder to reproduce than I initially thought. Attachments are connected to the issue, but it isn't consistently failing for every thread with attachments in it.

**Comment by Roensby** [comment]
The issue seems to revolve around https://github.com/InboxSDK/InboxSDK/blob/f0d94aa3ff236ce4288a33513e7834aa66538e78/src/platform-implementation-js/dom-driver/gmail/gmail-driver/show-custom-thread-list.ts#L295

This array can contain duplicates for some messages containing attachments.

A quickfix addresses the issue:
```typescript
const completedIDPairs = (await Promise.all(idPairsWithRFC.map(pair => idPairWithRFCToCompletedIDPair(driver, pair, findIdFailure)))).filter(Boolean);
const uniqueCompletedIDPairs = completedIDPairs.filter((element, index) => {
  return completedIDPairs.findIndex(searchFor => searchFor.gtid === element.gtid) === index
});
```

However, this change may cause a later call to determine if a reorder is needed to be `false`, since the `extractedThreads` list and the `extractedThreadsInCompletedIDPairsOrder` list can now be equal. It seems the variable `doesNeedReorder` was relying on duplicates in `completedIDPairs`. So the check to determine if a reorder is needed will need to be rewritten. 

During debug, if I force a reorder by setting `doesNeedReorder` to `true`, the issue is resolved: the custom list renders as expected every time, even with "problematic" messages containing attachments.

**Comment by Roensby** [comment]
Suggested fix: https://github.com/InboxSDK/InboxSDK/pull/1145

No changes to `doesNeedReorder` calculation needed.

**Comment by Macil** [comment]
Fixed in 2.1.38. Thanks for investigating this!
